web/Services/LandlordService.cs

```diff
@@ -85,9 +84,16 @@ await using (var transaction = await _dbContext.Database.BeginTransactionAsync(I
         return _dbContext.Landlords.SingleOrDefaultAsync(l => l.Id == id);
     }
 
-    public List<PropertyDbModel> GetListOfProperties(int landlordId)
+    public async Task ApproveLandlord(int landlordId,  BricksAndHeartsUser user)
     {
-        return _dbContext.Properties
-            .Where(p => p.LandlordId == landlordId).ToList();
+        var landlord = await GetLandlordIfExistsFromId(landlordId);
+        if (landlord!.CharterApproved)
+        {
+            throw new Exception("Landlord already approved");
+        }

```

Comment:
The result does actually change slightly - it would update the ApprovalTime, and possibly also the ApprovalAdminId, if a landlord was re-approved. Ben pointed out this slight difference and suggested it would be worth just adding this quick check, which made sense to me.

Reply:
@emilypatsko you've been making too many APIs, you can't `PUT` from an HTML form 😛